lektor/publisher.py

```diff
@@ -651,18 +651,18 @@ def detect_target_branch(self, target_url):
         return branch
 
     @staticmethod
-    def _check_error(line):
+    def _check_error(cmd, line):
         """
         Check that line does not contain a git error keyword.
         Check if a line (string) is interpreted as an error using a list of error related words.
-        If it is interpreted as an error, an exception is raised.
+        If interpreted as an error, an exception is raised
+        :param cmd: Lektor Command
         :param line: String. Line to evaluate. Ie, 'error: src refspec gh-pages does not match any'
         """
-        error_substrings = ['fatal:', 'error:']
-
-        if any(substring in line.lower() for substring in error_substrings):
+        if cmd.wait() == 1:
```

Comment:
This should probably be != 0 @arendondiosa ?

Reply:
Also since you are now waiting, the logic of the yield keeps working alright?

Since the command fails it might not be able to yield lines one by one. In this case we might need to change the for for something that just waits, does the check and then prints all the lines at once